### PR TITLE
feat: Add domain to Hubspot company

### LINKED ISSUE #3626: Hubspot: send domain name with Company to Hubspot
When we create a company in Hubspot, we should also send the domain name. 

image

Looks like the field is a built in Hubspot field so we should be able to provide it in the JSON we send across when we create the company. The domain name should be taken from the email address of the user creating / joining the organisation. 

If the user has a generic email address (gmail.com, hotmail.com, etc.) then we should try to avoid setting it where possible.

### PR DESCRIPTION
**Pull request by zachaysan**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Fixes https://github.com/Flagsmith/flagsmith/issues/3626

Simple change to introduce adding domain to the Hubspot company model.

## How did you test this code?

Since this change was purely against their API, I elected to just test it manually which was successful.

### CODE REVIEW

File: api/app/settings/common.py [hunk 1]
```diff
@@ -1033,7 +1033,33 @@
 ENABLE_HUBSPOT_LEAD_TRACKING = env.bool("ENABLE_HUBSPOT_LEAD_TRACKING", False)
 HUBSPOT_IGNORE_DOMAINS = env.list("HUBSPOT_IGNORE_DOMAINS", [])
 HUBSPOT_IGNORE_DOMAINS_REGEX = env("HUBSPOT_IGNORE_DOMAINS_REGEX", "")
-
+HUBSPOT_IGNORE_ORGANISATION_DOMAINS = env.list(
+    "HUBSPOT_IGNORE_ORGANISATION_DOMAINS",
+    [
+        "gmail.com",
+        "hotmail.com",
```
Comment (kyle-ssg) [inline api/app/settings/common.py:1049]: (being nosey) hotmail is duplicated, this is a list I use to filter out big-signups if it helps https://github.com/Kikobeats/free-email-domains/blob/master/domains.json#L1

Comment (zachaysan) [reply api/app/settings/common.py:1049]: Is this huge list being used on something at Flagsmith or for some other project? I'd just be wary of adding so many domains which are largely dormant. 

Edit: And thanks for spotting the duplication, I've removed it and sorted the list for easier visual verification and lookup.

File: api/integrations/lead_tracking/hubspot/lead_tracker.py [hunk 2]
```diff
@@ -54,18 +54,24 @@ def create_lead(self, user: FFAdminUser, organisation: Organisation) -> None:
...
         """
         if getattr(organisation, "hubspot_organisation", None):
             return organisation.hubspot_organisation.hubspot_id
 
-        response = self.client.create_company(name=organisation.name)
+        if user.email_domain in settings.HUBSPOT_IGNORE_ORGANISATION_DOMAINS:
+            domain = None
+        else:
+            domain = user.email_domain
+        response = self.client.create_company(name=organisation.name, domain=domain)
```
Comment (matthewelwell) [inline api/integrations/lead_tracking/hubspot/lead_tracker.py]: We should update the unit tests for this logic.

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: api/integrations/lead_tracking/hubspot/lead_tracker.py (hunk 2 above)
Comment (zachaysan) [reply api/integrations/lead_tracking/hubspot/lead_tracker.py]: Ok altered one existing test to capture when it is set and created a new test for when the domain name is in the filter list.

File: api/app/settings/common.py (hunk 1 above)
Comment (kyle-ssg) [reply api/app/settings/common.py:1049]: Anther project

Comment (kyle-ssg) [reply api/app/settings/common.py:1049]: Another project

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**